sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.institutions.controller.ts

```diff
@@ -186,4 +204,25 @@ export class EducationProgramInstitutionsController extends BaseController {
       userToken.authorizations.institutionId,
     );
   }
+
+  /**
+   * Checks if the user is authorized to create or modify programs for the institution.
+   * User should have a user type different from read-only user.
+   * @param userToken user token.
+   * @throws ForbiddenException if the user is not authorized.
+   */
+  private checkAuthorization(
```

Comment:
This method should be moved to the controller service to keep the controller free of any code other than the endpoints themselves.